web/public/js/controllers/user-profile-controller.js

```diff
@@ -273,79 +292,43 @@ function cdUserProfileCtrl($scope, $rootScope, $state, $window, auth, cdUsersSer
-            async.mapSeries($scope.profile.children, function(child, doneChild){
-              childrenCopy.name = child.name;
-              childrenCopy.alias = child.alias;
-              childrenCopy.dob = child.dateOfBirth;
-              childrenCopy.gender = child.gender;
-              childrenCopy.email = child.email;
-              childrenCopy.parents = [profileCopy.id];
-              if(getAge(child.dateOfBirth) >=13){
-                childrenCopy.userTypes = ['attendee-o13'];
-              } else {
-                childrenCopy.userTypes = ['attendee-u13'];
-              }
-              saveYouthViaParent(childrenCopy, doneChild);
-            }, function(err, results){
-              callback(err, results);
-            });
-          }else{
-            callback(null, []);
+          if ($state.params.parentId && !_.includes(profileCopy.parents, $state.params.parentId)) {
+            profileCopy.parents.push($state.params.parentId);
           }
+          saveYouthViaParent(profileCopy, callback);
+        } else {
+          saveDirect(profileCopy, callback);
         }
-      ], function (err, results) {
-          results = _.flatten(results);
-          var messages = [];
-          //  Server error
-          if (err) {
-            return alertService.showError($translate.instant('An error has occurred while saving profile'));
-          } else { // Functional errors
-            var errorous = false;
-            _.each( results, function (result) {
-              if(result && result.error){
-                var error_string = "";
-                error_string = result.error === 'nick-exists' ? $translate.instant('Alias already exists.') : result.error;
-                messages.push($translate.instant('An error has occurred while saving youth profile') + ': ' + error_string);
-                errorous = true;
-              }
-              if (result && result.ok === false) {
-                messages.push($translate.instant(result.why));
-                errorous = true;
-              } else {
-                messages.push($translate.instant('Profile(s) have been saved successfully'));
-              }
-            });
-            var message = _.uniq(messages).join('</br>');
-            if(errorous){
-              return alertService.showError(message);
+      },
+      function (callback) {
+        if ($scope.profile.children !== null) {
+          localStorage.setItem('children', 'true');
+          async.mapSeries($scope.profile.children, function (child, doneChild) {
+            childrenCopy.name = child.name;
+            childrenCopy.alias = child.alias;
+            childrenCopy.dob = child.dateOfBirth;
+            childrenCopy.gender = child.gender;
+            childrenCopy.email = child.email;
+            childrenCopy.parents = [profileCopy.id];
+            if (getAge(child.dateOfBirth) >= 13) {
+              childrenCopy.userTypes = ['attendee-o13'];
```

Comment:
Not sure what happened here (and the ~50 next lines), but you mixed up the final callback with one of the series actions. You also removed the conditional assignation to u13 userType.
Why? :p